Ports "Smooth movement - "TG coders suck" edition" from yogstation 13 (again)

BrumalLake opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Ports [Yogstation#8132](https://github.com/yogstation13/Yogstation/pull/8132)
Brings over the buttery movement, and makes mobs with slow movement speeds glide instead of jump from tile to tile
Small modifications taken from [tgstation#52515](https://github.com/tgstation/tgstation/pull/52515)
Moving at high speeds no longer looks like you're teleporting
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Movement feels a lot smoother and moving things wont be as stupid hard to click
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: monster860, actioninja, lordScrubling
add: Smooth movement from yogstation
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
ctrl v

ToasterBan commented:
someone already tried to port this I think? #1513

BrumalLake commented:
what went wrong
oh, a vote
maybe people have changed since then

Commit:
less fps

BrumalLake commented:
This is immediately going to get closed because of the vote isnt it
dont know why people hate this

utilbird commented:
@lordScrubling Because it makes the game uncomfortable to play at higher FPS

Sinestia commented:
> 
> 
> @lordScrubling Because it makes the game uncomfortable to play at higher FPS

How? I'd wager it'd make it better for ultra high FPS setups. 

This is probably just boomer propaganda

eLeCtrOssSnake commented:
Smooth movement is one of the reasons I don’t like playing on TG. It feels childish.

eLeCtrOssSnake commented:
> > @lordScrubling Because it makes the game uncomfortable to play at higher FPS
> 
> How? I'd wager it'd make it better for ultra high FPS setups.
> 
> This is probably just boomer propaganda

No it actually slows it down in a weird looking way.

SinfulBliss commented:
It's weird at first but it's way better and more logical after you warm up to it.

utilbird commented:
> How? I'd wager it'd make it better for ultra high FPS setups.
> This is probably just boomer propaganda

#1513
Even monster themselves said it 😛

utilbird commented on code/__DEFINES/movement.dm:
```diff
@@ -0,0 +1,16 @@
+// Originally a really stupid /tg/ var that sucked and was really bad and caused it to look horrible. Now it's a way of compensating for time dilation
+GLOBAL_VAR_INIT(glide_size_multiplier, 1.0)
+
+///Broken down, here's what this does:
+/// divides the world icon_size (32) by delay divided by ticklag to get the number of pixels something should be moving each tick.
+/// The division result is given a min value of 1 to prevent obscenely slow glide sizes from being set
+/// Then that's multiplied by the global glide size multiplier. 1.25 by default feels pretty close to spot on. This is just to try to get byond to behave.
+/// The whole result is then clamped to within the range above.
+/// Not very readable but it works
+#define DELAY_TO_GLIDE_SIZE(delay) (CLAMP(((32 / max((delay) / world.tick_lag, 1)) * GLOB.glide_size_multiplier), MIN_GLIDE_SIZE, MAX_GLIDE_SIZE))
```
This is quite expensive

PowerfulBacon replied:
its literally just multiplication, although it could probably be cached when the move delay changes

utilbird commented:
I believe tgstation/tgstation#52515 fixed some of these issues though

Commit:
Merge remote-tracking branch 'Upstream/master' into smooth-movement

Commit:
tg 52515

BrumalLake commented:
52515 didnt change anything much, just made the diagonal movement checks simpler to understand, removed the sloth's previous glide and changed something about orbiting

Commit:
Merge remote-tracking branch 'Upstream/master' into smooth-movement

BrumalLake commented:
bruh some dude put a single space after some if statements and created a conflict

Commit:
Merge remote-tracking branch 'Upstream/master' into smooth-movement

Commit:
prep

ike709 commented:
We'll try it for a few rounds.

Commit:
Merge remote-tracking branch 'Upstream/master' into smooth-movement

Commit:
prep

Commit:
Merge remote-tracking branch 'Upstream/master' into smooth-movement

francinum commented:
This feels like shit at 40 and gets sickening the higher you go.

ike709 commented:
I personally like it but it does make me feel physically ill yeah.

BrumalLake commented:
doesn't seem to work very well with 50% td which is average it seems
